### PR TITLE
Deprecate ARobotVehicle & Doc update

### PR DESCRIPTION
**Pull request by yuokamoto**
Deprecate `ARobotVehicle`, all child robot classes will inherit directly from `ARRBaseRobot`
Doc https://rapyutasimulationplugins.readthedocs.io/en/doc_update/

- update documentation: add Robots and Getting Started
- replace ROS2 with ROS 2
- add InitPropertiesFromJson and bMobile

### CODE REVIEW

File: Source/RapyutaSimulationPlugins/Public/Robots/RRBaseRobot.h [hunk 1]
```diff
@@ -443,6 +444,18 @@ class RAPYUTASIMULATIONPLUGINS_API ARRBaseRobot : public ARRBaseActor
     UFUNCTION(BlueprintImplementableEvent, BlueprintCallable)
     void BPConfigureMovementComponent();
 
+    //! Mobile robot or not. If this is false, movecomponent=nullptr and ROS 2 odom and cmd_vel interface are disabled.
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    bool bMobile = true;
```
Comment (Tadinu) [inline Source/RapyutaSimulationPlugins/Public/Robots/RRBaseRobot.h:449]:
```suggestion
    //! Vehicle robot or not. If this is false, movecomponent=nullptr and ROS 2 odom and cmd_vel interface are disabled.
    bool bVehicle = true;
```
just to make it consistent with `ATurtlebotBurgerVehicle`, etc. & separate itself from `AActor::IsRootComponentMovable()`

**Review by Tadinu — APPROVED** [review]

File: Source/RapyutaSimulationPlugins/Public/Robots/RRBaseRobot.h (hunk 1 above)
Comment (Tadinu) [inline Source/RapyutaSimulationPlugins/Public/Robots/RRBaseRobot.h]:
```suggestion
    //! Mobile robot or not. If this is false, movecomponent=nullptr and ROS 2 odom and cmd_vel interface are disabled.
    bool bMobileRobot = true;
```
just to make its meaning clear & separated from `AActor::IsRootComponentMovable()`

File: Source/RapyutaSimulationPlugins/Public/Core/RRNetworkPlayerController.h [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 /**
  * @brief Network Player controller provides functionality for client-server. Major functionalites are
- * - [UROS2NodeComponent](https://rclue.readthedocs.io/en/devel/doxygen_generated/html/d1/d79/_r_o_s2_node_component_8h.html),  #URRROS2ClockPublisher,  #URRROS2SimulationStateClient are created for each client to provide ROS2 services which are provided by #ARRROS2GameMode in standalone game.
+ * - [UROS2NodeComponent](https://rclue.readthedocs.io/en/devel/doxygen_generated/html/d1/d79/_r_o_s2_node_component_8h.html),  #URRROS2ClockPublisher,  #URRROS2SimulationStateClient are created for each client to provide ROS 2 services which are provided by #ARRROS2GameMode in standalone game.
  * - Clock sync between server and the clients with delay compensation
  * - RPC call to sync Robot movements between serer and clients.
```
Comment (Antho1426RR) [inline Source/RapyutaSimulationPlugins/Public/Core/RRNetworkPlayerController.h]:
```suggestion
 * - RPC call to sync Robot movements between server and clients.
```

File: Source/RapyutaSimulationPlugins/Public/Core/RRROS2GameMode.h [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /**
  * @file RRROS2GameMode.h
- * @brief ROS2 GameMode which have Clock publisher and ROS2 services to interact with ROS2.
+ * @brief ROS 2 GameMode which have Clock publisher and ROS 2 services to interact with ROS2.
```
Comment (Antho1426RR) [inline Source/RapyutaSimulationPlugins/Public/Core/RRROS2GameMode.h]:
```suggestion
 * @brief ROS 2 GameMode which have Clock publisher and ROS 2 services to interact with ROS 2.
```

File: Source/RapyutaSimulationPlugins/Public/Core/RRROS2GameMode.h [hunk 4]
```diff
@@ -23,7 +23,7 @@ class URRROS2ClockPublisher;
 DECLARE_MULTICAST_DELEGATE(FRROnROS2Initialized);
 
 /**
- * @brief ROS2 GameMode which have Clock publisher and ROS2 services to interact with ROS2.
+ * @brief ROS 2 GameMode which have Clock publisher and ROS 2 services to interact with ROS2.
```
Comment (Antho1426RR) [inline Source/RapyutaSimulationPlugins/Public/Core/RRROS2GameMode.h]:
```suggestion
 * @brief ROS 2 GameMode which have Clock publisher and ROS 2 services to interact with ROS 2.
```

File: docs/source/overview.rst [hunk 5]
```diff
@@ -1,49 +1,131 @@
...
+    ARRBaseRobot has only basic interface such as `/cmd_vel`, `/odom`. `/joint_states`.
+    Please create child class and add necessary ROS 2 APIs.
 
-| \* This provide typical way of creating robot. You should be able to create Actor which has ROS interfaces with arbitrary architecture.
-| \* Following the Pawn and AIController structure in Unreal Engine, `Ref <https://docs.unrealengine.com/4.27/en-US/InteractiveExperiences/Framework/Pawn/>`_.
+- Spawning from ROS 2 with parameters
+    ARRBaseRobot and URRRobotROS2Interface can be spawnd from `/SpawnEntity <https://github.com/rapyuta-robotics/UE_msgs/blob/devel/srv/SpawnEntity.srv>`_ 
+    service. You can pass Robot Name, Namespace and etc.
 
+    SpawnEntity.srv also has filed named `json_parameters` to pass random parameters.
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
    SpawnEntity.srv also has a field named `json_parameters` to pass random parameters.
```

File: docs/source/overview.rst [hunk 6]
```diff
@@ -1,49 +1,131 @@
...
+    Please add your own custom parser in child class by overwriting 
+    `InitPropertiesFromJSON() <doxygen_generated/html/df/d13/class_a_r_r_base_robot.html#a214c5936450e3b17dffaad40e944bea6>`_ 
+    which is called in 
+    the `PreInitializeComponents() <https://docs.unrealengine.com/5.1/en-US/unreal-engine-actor-lifecycle/>`_
+    
+- Distributed simulation
+    ARRBaseRobot has a seting to be used for client-server. Please check :doc:`distributed_simulation` for detailed information.
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
    ARRBaseRobot has a setting to be used for client-server. Please check :doc:`distributed_simulation` for detailed information.
```

File: docs/source/overview.rst [hunk 7]
```diff
@@ -1,49 +1,131 @@
...
+- `ARRBaseRobot <doxygen_generated/html/df/d13/class_a_r_r_base_robot.html>`_
+    Base Robot class which has basic ROS 2 topic interfaces such as Twist msg, JointState and etc via ROS2Interface.
+    - `URRROS2BaseSensorComponent <doxygen_generated/html/d0/d58/class_u_r_r_r_o_s2_base_sensor_component.html>`_ 
+        Base class of ROS 2 sensors. Other ROS 2 sensor should be child class of this class, e.g. lidar, camera, etc. 
+
+        URRROS2BaseSensorComponent has `URRROS2BaseSensorPublisher <doxygen_generated/html/d5/d69/class_u_r_r_r_o_s2_base_sensor_publisher.html>`_ 
+        which publish Sensor topic and is initialized from URRROS2Interface.
+
+    - `URobotVehicleMovementComponent <doxygen_generated/html/d7/d01/class_u_robot_vehicle_movement_component.html>`_ 
+        Base class of Robot movement, which is controled by
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
        Base class of Robot movement, which is controlled by 
```

File: docs/source/overview.rst [hunk 8]
```diff
@@ -1,49 +1,131 @@
...
 
 GameMode
 -----------------
 
-The GameMode start all the simulation components, mainly the Simulation State and its associated ROS2 Node.
+The GameMode start all the simulation components, mainly the Simulation State and its associated ROS 2 Node.
+
+- `ASimulationState <doxygen_generated/html/d2/dde/class_a_simulation_state.html>`_ 
+    
+    Simulation component, which has umplementation of ROS 2 services and communicate with URRROS2SimulationstateClient.
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
    Simulation component, which has implementation of ROS 2 services and communicate with URRROS2SimulationstateClient.
```

File: docs/source/overview.rst [hunk 9]
```diff
@@ -1,49 +1,131 @@
...
+    
+    ASimulationState can not control all actors, but it only manages actors in 
+    `Entities <doxygen_generated/html/d2/dde/class_a_simulation_state.html#a814e73a90bfbe5e01973e7b04552dfa1>`_  
+    . Actors in the level at BeginPlay and actors spawned by ROS 2 services are added automatically.
+    You need to manually call `ServerAddEntity() <doxygen_generated/html/d2/dde/class_a_simulation_state.html#a26555da061d66a6b31f5c42053708119>`_ 
+    if you want to control actors that are not spawned from ASimulationState at runtime.
+  
+    ASimulationState can not spawn all UE actor classes, but only  
+    `SpawnableEntitiTypes <doxygen_generated/html/d2/dde/class_a_simulation_state.html#a236405852be150d955a4736a323cb514>`_
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
    `SpawnableEntityTypes <doxygen_generated/html/d2/dde/class_a_simulation_state.html#a236405852be150d955a4736a323cb514>`_ 
```

File: docs/source/overview.rst [hunk 10]
```diff
@@ -1,49 +1,131 @@
...
-| \* `ASimulationState <doxygen_generated/html/d2/dde/class_a_simulation_state.html>`_ has Entities and only able to manipulate actors Entities.
-| \* You need to use specific method to add actors to the variables if you want to manipulate from ROS2 services. Actors in the level at beginplay and spawned actor by ROS2 services are added automatically. You can change spawnable entities in level BP, by overwriting game mode and etc.
+- `URRROS2ClockPublisher <doxygen_generated/html/d5/dc2/class_u_r_r_r_o_s2_clock_publisher.html>`_ 
+  
+    ROS 2 publisher class publishes `/clock` topic at every tick in the simulation. 
+  
+    In the example `turtlebot3-UE <https://github.com/rapyuta-robotics/turtlebot3-UE>`_ project,
+    It is set to use
+    `URRLimitRTFFixedSizeCustomTimeStep <doxygen_generated/html/d0/d4d/class_u_r_r_limit_r_t_f_fixed_size_custom_time_step.html>`_
+    which enabes running simulation with fixed timestep while limiting the RTF(Real Time Factor).
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
    which enables running simulation with fixed timestep while limiting the RTF(Real Time Factor).
```

File: docs/source/overview.rst [hunk 11]
```diff
@@ -1,49 +1,131 @@
...
 
+    The turtlebot3-UE project is set to update `DefaultEngine.ini`` with environment variable from 
+    `DefaultEngineBase.ini <https://github.com/rapyuta-robotics/turtlebot3-UE/blob/devel/Config/DefaultEngineBase.ini>`_
+    . The frame rate and max RTF are set from environment varibale `FIXED_FRAME_RATE` and `TARGET_RTF`.
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
    . The frame rate and max RTF are set from environment variable `FIXED_FRAME_RATE` and `TARGET_RTF`.
```

File: docs/source/overview.rst [hunk 12]
```diff
@@ -56,7 +138,7 @@ The GameMode start all the simulation components, mainly the Simulation State an
      - `SetEntityState.srv <https://github.com/rapyuta-robotics/UE_msgs/blob/devel/srv/SetEntityState.srv>`_
      - et the actor state.
    * - /GetEntityState
-     - `Attach.srv <https://github.com/rapyuta-robotics/UE_msgs/blob/devel/srv/Attach.srv>`_
+     - `GetEntityState.srv <https://github.com/rapyuta-robotics/UE_msgs/blob/devel/srv/GetEntityState.srv>`_
      - return the actor actor state
```
Comment (tetorea) [inline docs/source/overview.rst]:
```suggestion
     - return the actor state
```

File: docs/source/robots/mobile_robot.rst [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+     - Laser scan msg
+
+
+Turtlebot3 Burger
+""""""""""""""""""
+
+Turtlebot3 Burger Actors are examples of robots implemented in C++. 
+All components, joints, etc are defined in C++.
+
+Even if it is implemented in C++, we defined Blueprint child class to change parametrs and assign meshes.
```
Comment (tetorea) [inline docs/source/robots/mobile_robot.rst]:
```suggestion
Even if it is implemented in C++, we defined Blueprint child class to change parameters and assign meshes.
```

File: docs/source/robots/mobile_robot.rst [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+Turtlebot3 Waffle Actors are examples of robots implemented in BP.
+Both are child class of ARRBaseRobot and all additional components, joints, etc are defined in BP.
+
+- `BP_TurtlebotWaffleVehicle <https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/devel/Content/Robots/Turtlebot3/Kinematics/BP_TurtlebotWaffleVehicle.uasset>`_
+    Example kinematic robot implementation in BP. 
+    
+    This is child class of ARRBaseRobot and is added 
+    `SkeletalMesh <https://docs.unrealengine.com/4.27/en-US/API/Runtime/Engine/Components/USkeletalMeshComponent/>`_ 
+    and `URR2DLidarComponent  <../doxygen_generated/html/d4/d87/class_u_r_r2_d_lidar_component.html>`_
+    simillar as ATurtlebotBurgerVehicle in C++.
```
Comment (tetorea) [inline docs/source/robots/mobile_robot.rst]:
```suggestion
    similar as ATurtlebotBurgerVehicle in C++.
```

File: docs/source/robots/mobile_robot.rst [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+    .. image:: ../images/turtlebot3_waffle_vehicle.png
+
+- `BP_TurtlebotWaffle <https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/devel/Content/Robots/Turtlebot3/Physics/BP_TurtlebotWaffle.uasset>`_
+    Example physics robot implementation in BP. 
+
+    This is child class of ARRBaseRobot and is added 
+    `UStaticMeshComponent <https://docs.unrealengine.com/4.27/en-US/API/Runtime/Engine/Components/UStaticMeshComponent/>`_, 
+    `UPhysicsConstraintComponent <https://docs.unrealengine.com/4.26/en-US/API/Runtime/Engine/PhysicsEngine/UPhysicsConstraintComponent/>`_, 
+    `UDifferentialDriveComponent <../doxygen_generated/html/db/df5/class_u_differential_drive_component.html>`_, 
+    and `URR2DLidarComponent  <../doxygen_generated/html/d4/d87/class_u_r_r2_d_lidar_component.html>`_ simillar as ATurtlebotBurger in C++.
```
Comment (tetorea) [inline docs/source/robots/mobile_robot.rst]:
```suggestion
    and `URR2DLidarComponent  <../doxygen_generated/html/d4/d87/class_u_r_r2_d_lidar_component.html>`_ similar as ATurtlebotBurger in C++.
```

File: docs/source/robots/mobile_robot.rst [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+
+    .. image:: ../images/turtlebot3_waffle.png
+
+
+    To configure MoveComponent in BP, 
+    `VehicleMoveComponentClass <../doxygen_generated/html/df/d13/class_a_r_r_base_robot.html#aa69278b89215d02dd07da74b6feb83f3>`_
+    is set as None, then, VehiclMoveComponent is not configure in C++ function.
```
Comment (tetorea) [inline docs/source/robots/mobile_robot.rst]:
```suggestion
    is set as None, then, VehicleMoveComponent is not configured in C++ function.
```

File: docs/source/index.rst [hunk 17]
```diff
@@ -2,11 +2,11 @@ Welcome to RapyutaSimulationPlugins's documentation!
 ====================================================
 
 **RapyutaSimulationPlugins** includes robot common assets to build
-`ROS2 <https://docs.ros.org/en/foxy/index.html#>`_ simulator with
+`ROS 2 <https://docs.ros.org/en/foxy/index.html#>`_ simulator with
 `Unreal Engine <https://www.unrealengine.com/en-US/?lang=en-US>`_.
 
 RapyutaSimulationPlugins provide UE components
-which provide feature to control Actor from ROS2, output simulated sensor to ROS2 and etc.
+which provide feature to control Actor from ROS2, output simulated sensor to ROS 2 and etc.
```
Comment (Antho1426RR) [inline docs/source/index.rst]:
```suggestion
which provide feature to control Actor from ROS 2, output simulated sensor to ROS 2 and etc.
```

File: docs/source/robots/mobile_robot.rst [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+--------------------------
+
+To create custom mobile robot
+
+1. Create child class of ARRBaseRobot
+    1. Overwrite default parameters.
+    2. Configure meshes and physics constraints in BP.
+    3. If you want to dynamically spawn robots and pass random parameters, overwrite `InitPropertiesFromJSON() <../doxygen_generated/html/df/d13/class_a_r_r_base_robot.html#a214c5936450e3b17dffaad40e944bea6>`_ 
+2. Create child class of URobotVehicleMovementComponent if you want custom movement behaviour.
+    1. Configure movecomponent in your Robot Class simillar as BP_TurtlebotWaffle.s
```
Comment (tetorea) [inline docs/source/robots/mobile_robot.rst]:
```suggestion
    1. Configure movecomponent in your Robot Class similar as BP_TurtlebotWaffle.
```

File: docs/source/robots/robot_arm.rst [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+Custom Robot Creation TIPS
+--------------------------
+
+To create custom mobile robot
+
+1. Create child class of ARRBaseRobot
+    1. Overwrite default parameters.
+        1. Set `bMobileRobot <../doxygen_generated/html/df/d13/class_a_r_r_base_robot.html#a315bb15cfa368ec652622e27ae150c6a>`_ false.
+    2. Configure meshes and joints in BP
+    3. Configure construction script to set joint and link relations simillar as BP_KinematicSimpleArm.
```
Comment (tetorea) [inline docs/source/robots/robot_arm.rst]:
```suggestion
    3. Configure construction script to set joint and link relations similar as BP_KinematicSimpleArm.
```

**Review by tetorea — APPROVED** [review]

**Review by Antho1426RR — APPROVED** [review]

**Pull request merged**